index.d.ts

```diff
@@ -0,0 +1,128 @@
+declare module 'fluture' {
+  interface ResolveFunction<T> {
+    (value: T): void
+  }
+  interface RejectFunction<S> {
+    (error: S): void
+  }
+
+  interface FunctionReturning<T> {
+    (...t: any[]): T
+  }
+  abstract class Future<T, S> {
+    done<T, S> (fn: (err: S, value: T) => void): Future<T, S>
+
+    fork (reject: RejectFunction<S>, resolve: ResolveFunction<T>): void
+  }
+
+  function after<T, S> (duration: number, value: string): Future<T, S>
+
+  function and<A, B> (a: Future<A, any>, b: Future<B, any>): Future<A | B, any>
```

Comment:
I would like to avoid mixed types, and force users to use consistent types. We can avoid a mixed type in this function by making the input more strict:

```ts
function and<L, R> (a: Future<L, any>, b: Future<L, R>): Future<L, R>
```